### PR TITLE
BUG: Fix .hist and .plot.hist when passing existing figure (#37278)

### LINKED ISSUE #37278: BUG: Undocumented, Inconsistent matplotlib figure usage (Series.hist vs Series.plot.hist)
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
# Your code here
from pandas import DataFrame
import matplotlib.pyplot as plt

d = DataFrame({'y1':[0,1,2,3], 'y2':[3,4,5,6]}, index=[0,1,2,3])

# A new matplotlib figure is created
print(plt.gcf().number)
d.hist()
print(plt.gcf().number)  # new figure is created, current matplotlib figure is the new one

# Can NOT assign a figure to a dataframe plot
figure = plt.figure()
print(figure.number)
print(plt.gcf().number)
d.hist(figure=figure)  # Throws an exception
print(plt.gcf().number)  # new figure is still created! current matplotlib figure is the new one

# Can assign a figure to a Series histogram
figure1 = plt.figure()
figure2 = plt.figure()  # Current figure is always the last one created
print(figure1.number)
print(figure2.number)
print(plt.gcf().number)
d['y1'].hist(figure=figure1)
print(plt.gcf().number)  # Current figure stays the same
figure1.show() # This works

# Can NOT assign a figure to a Series histogram when using the .plot CachedAccessor
# However, it does not create a new figure
figure1 = plt.figure()
figure2 = plt.figure()  # Current figure is always the last one created
print(figure1.number)
print(figure2.number)
print(plt.gcf().number)
d['y1'].plot.hist(figure=figure1)  # Throws an exception!
print(plt.gcf().number)  # Current figure stays the same
```

#### Problem description

It is currently impossible to assign a plot of a DataFrame to an existing figure.  The functionality is just broken.

The API to set a Series to an existing figure is almost completely broken (with the exception of the direct hist function).
There is a workaround in setting the global current figure `plt.figure(old_figure.number)`, but that does not work if `plt.Figure()` is used.

#### Expected Output

All three examples accept the `figure=` argument, and use it.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.8.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.4.0-51-generic
Version          : #56-Ubuntu SMP Mon Oct 5 14:28:49 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 1.1.3
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 44.0.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : 1.0.1
pymysql          : None
psycopg2         : 2.8.6 (dt dec pq3 ext lo64)
jinja2           : 2.11.2
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : 0.8.4
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.2
numexpr          : None
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.3
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by krsnik93**
- [x] closes #37278
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -476,6 +476,7 @@ Plotting
 
 - Bug in :meth:`DataFrame.plot` was rotating xticklabels when ``subplots=True``, even if the x-axis wasn't an irregular time series (:issue:`29460`)
 - Bug in :meth:`DataFrame.plot` where a marker letter in the ``style`` keyword sometimes causes a ``ValueError`` (:issue:`21003`)
+- Bug in :meth:`DataFrame.hist`, :meth:`DataFrame.plot.hist` and :meth:`Series.plot.hist` when passing an existing figure (:issue:`37278`)
```
Comment (arw2019) [inline doc/source/whatsnew/v1.2.0.rst:479]: this should go in enhancements I think. Something like
```
:meth:`DataFrame.hist`, :meth:`DataFrame.plot.hist` can now be called with an existing matplotlib figure object via added ``figure`` argument
```
Possibly add an example (not 100% that's needed though)

File: pandas/tests/plotting/test_hist_method.py [hunk 2]
```diff
@@ -152,6 +152,13 @@ def test_hist_with_legend_raises(self, by):
         with pytest.raises(ValueError, match="Cannot use both legend and label"):
             s.hist(legend=True, by=by, label="c")
 
+    def test_hist_with_passed_figure(self):
```
Comment (arw2019) [inline pandas/tests/plotting/test_hist_method.py:155]: I'd call this `test_hist_with_figure_argument`

File: pandas/tests/plotting/test_hist_method.py [hunk 3]
```diff
@@ -395,6 +402,13 @@ def test_hist_with_legend_raises(self, by, column):
         with pytest.raises(ValueError, match="Cannot use both legend and label"):
             df.hist(legend=True, by=by, column=column, label="d")
 
+    def test_hist_with_passed_figure(self):
```
Comment (arw2019) [inline pandas/tests/plotting/test_hist_method.py:405]: I'd call this `test_hist_with_figure_argument`

**Review by arw2019 — COMMENTED** [review]
Thanks @krsnik93 for the PR!

Not super familiar with this part of the code base but looks good - some comments re: whatsnew & tests

**Review by arw2019 — COMMENTED** [review]
@krsnik93 sorry we've been quiet on this - would you mind merging master & resolving conflicts (it's been a while)

looks good to me but I don't know this part of the code base well

**Comment by jbrockmendel** [comment]
@charlesdong1991 can you take a look

**Comment by jreback** [comment]
closing as stale, if you want to continue working, please ping.

**Pull request closed without merging**